Userland/Libraries/LibWeb/WebIDL/Promise.cpp

```diff
@@ -69,22 +70,40 @@ JS::NonnullGCPtr<Promise> create_rejected_promise(JS::Realm& realm, JS::Value re
 }
 
 // https://webidl.spec.whatwg.org/#resolve
-void resolve_promise(JS::VM& vm, Promise const& promise, JS::Value value)
+void resolve_promise(JS::Realm& realm, Promise const& promise, JS::Value value)
 {
+    auto& vm = realm.vm();
+
     // 1. If x is not given, then let it be the undefined value.
     // NOTE: This is done via the default argument.
 
     // 2. Let value be the result of converting x to an ECMAScript value.
 
+    // NOTE: Not part of the spec, but we need to have an execution context on the stack to call native functions.
+    auto& environment_settings_object = Bindings::host_defined_environment_settings_object(realm);
+    environment_settings_object.prepare_to_run_script();
```

Comment:
Have you filed a spec issue about this?

Reply:
Note that this has been moved to consume_body (to handle JSON parsing and promises) in Fetch/Body.cpp and also Fetch/FetchMethod.cpp.

I haven't filed a spec issue about this, I'm not 100% sure on the fix itself anyway as it feels like a wider issue.